core-plugins/src/main/java/co/cask/hydrator/plugin/realtime/sink/RealtimeTableSink.java

```diff
@@ -58,11 +59,53 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     Preconditions.checkArgument(!Strings.isNullOrEmpty(tableSinkConfig.getName()), "Dataset name must be given.");
     Preconditions.checkArgument(!Strings.isNullOrEmpty(tableSinkConfig.getRowField()),
                                 "Field to be used as rowkey must be given.");
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      validateInputAndOutputSchemaFields(inputSchema);
+
+      boolean rowFieldPresent = false;
+      for (Schema.Field field : inputSchema.getFields()) {
```

Comment:
this logic also looks like common and repeated logic. Could also move it to common as `validateFieldsAreSimple` method or something like that.